src/wallet/mod.rs

```diff
@@ -383,38 +405,59 @@ where
             (Some(rbf), _) => rbf.get_value(),
         };
 
+        let (fee_rate, mut fee_amount) = match params
+            .fee_policy
+            .as_ref()
+            .unwrap_or(&FeePolicy::FeeRate(FeeRate::default()))
+        {
+            //FIXME: see https://github.com/bitcoindevkit/bdk/issues/256
+            FeePolicy::FeeAmount(fee) => {
+                if let Some(previous_fee) = params.bumping_fee {
+                    if *fee < previous_fee.absolute {
+                        return Err(Error::FeeTooLow {
+                            required: previous_fee.absolute,
+                        });
+                    }
+                }
+                (FeeRate::from_sat_per_vb(0.0), *fee as f32)
+            }
+            FeePolicy::FeeRate(rate) => {
+                if let Some(previous_fee) = params.bumping_fee {
+                    let required_feerate = FeeRate::from_sat_per_vb(previous_fee.rate + 1.0);
+                    if *rate < required_feerate {
+                        return Err(Error::FeeRateTooLow {
+                            required: required_feerate,
+                        });
+                    }
+                }
+                (*rate, 0.0)
+            }
+        };
+
         let mut tx = Transaction {
             version,
             lock_time,
             input: vec![],
             output: vec![],
         };
 
-        let (fee_rate, mut fee_amount) = match builder
-            .fee_policy
-            .as_ref()
-            .unwrap_or(&FeePolicy::FeeRate(FeeRate::default()))
-        {
-            FeePolicy::FeeAmount(amount) => (FeeRate::from_sat_per_vb(0.0), *amount as f32),
-            FeePolicy::FeeRate(rate) => (*rate, 0.0),
-        };
-
-        // try not to move from `builder` because we still need to use it later.
-        let recipients = match &builder.single_recipient {
+        // try not to move from `params` because we still need to use it later.
```

Comment:
Perhaps just remove this comment, it does not seem to add any real value.

Reply:
I agree.